src/date/expand-pattern/augment-format.js

```diff
@@ -4,30 +4,53 @@ define([
 	"../../util/string/repeat"
 ], function( dateExpandPatternNormalizePatternType, datePatternRe, stringRepeat ) {
 
-return function( requestedSkeleton, bestMatchFormat ) {
-	var i, j, matchedType, matchedLength, requestedType, requestedLength,
+function expandBestMatchFormat(skeletonWithoutFractionalSeconds, bestMatchFormat) {
+	var i, j, bestMatchFormatParts, matchedType, matchedLength, requestedType,
+		requestedLength, requestedSkeletonParts,
 
 		// Using an easier to read variable.
 		normalizePatternType = dateExpandPatternNormalizePatternType;
 
-	requestedSkeleton = requestedSkeleton.match( datePatternRe );
-	bestMatchFormat = bestMatchFormat.match( datePatternRe );
+	requestedSkeletonParts = skeletonWithoutFractionalSeconds.match( datePatternRe );
+	bestMatchFormatParts = bestMatchFormat.match( datePatternRe );
 
-	for ( i = 0; i < bestMatchFormat.length; i++ ) {
-		matchedType = bestMatchFormat[i].charAt( 0 );
-		matchedLength = bestMatchFormat[i].length;
-		for ( j = 0; j < requestedSkeleton.length; j++ ) {
-			requestedType = requestedSkeleton[j].charAt( 0 );
-			requestedLength = requestedSkeleton[j].length;
+	for ( i = 0; i < bestMatchFormatParts.length; i++ ) {
+		matchedType = bestMatchFormatParts[i].charAt( 0 );
+		matchedLength = bestMatchFormatParts[i].length;
+		for ( j = 0; j < requestedSkeletonParts.length; j++ ) {
+			requestedType = requestedSkeletonParts[j].charAt( 0 );
+			requestedLength = requestedSkeletonParts[j].length;
 			if ( normalizePatternType( matchedType ) === normalizePatternType( requestedType ) &&
 				matchedLength < requestedLength
 			) {
-				bestMatchFormat[i] = stringRepeat( matchedType, requestedLength );
+				bestMatchFormatParts[i] = stringRepeat( matchedType, requestedLength );
 			}
 		}
 	}
 
-	return bestMatchFormat.join( "" );
+	return bestMatchFormatParts.join( "" );
+}
+
+// See: http://www.unicode.org/reports/tr35/tr35-dates.html#Matching_Skeletons
+return function( requestedSkeleton, bestMatchFormat, decimalSeparator ) {
```

Comment:
It's good except for our build process, which assumes all files have one export (and no outer code). In order to have something different, we need an update the [onBuildWrite](https://github.com/globalizejs/globalize/blob/master/Gruntfile.js#L153), specifically tell it this file is a "[type b](https://github.com/globalizejs/globalize/blob/master/Gruntfile.js#L282)".